### PR TITLE
Multi-stage pipeline introspection documentation

### PR DESCRIPTION
**Pull request by edaena**

### CODE REVIEW

**Review by sarath-p — APPROVED** [review]
Looks good.

File: docs/service-introspection-onboarding.md [hunk 1]
```diff
@@ -149,32 +149,59 @@ by service introspection.
...
-./spk-linux deployment create  -n $(ACCOUNT_NAME) -k $(ACCOUNT_KEY) -t $(TABLE_NAME) -p $(PARTITION_KEY)  --p2 $(Release.ReleaseId) --hld-commit-id $latest_commit --env $(Release.EnvironmentName) --image-tag $(Build.BuildId)
+latest_commit=$(git rev-parse --short HEAD) VERSION_TO_DOWNLOAD=$(curl -s
+"https://api.github.com/repos/CatalystCode/spk/releases/latest" | grep
+"tag_name" | sed -E 's/.*"([^"]+)".*/\1/') echo "Downloading SPK version
+$VERSION_TO_DOWNLOAD" && wget
+"https://github.com/CatalystCode/spk/releases/download/$VERSION_TO_DOWNLOAD/spk-linux"
+chmod +x ./spk-linux ./spk-linux deployment create  -n $(ACCOUNT_NAME) -k
+$(ACCOUNT_KEY) -t $(TABLE_NAME) -p $(PARTITION_KEY)  --p2 $(Release.ReleaseId)
+--hld-commit-id $latest_commit --env $(Release.EnvironmentName) --image-tag
+$(Build.BuildId)
```
Comment (samiyaakhtar) [inline docs/service-introspection-onboarding.md]: As far as I can see, all this diff is doing is adding extra lines between the yaml, which will fail the bash task. You might want to fix the new lines. Other than the new lines, i don't see a diff here

File: docs/service-introspection-onboarding.md [hunk 2]
```diff
@@ -149,32 +149,59 @@ by service introspection.
...
+latest_commit=$(git rev-parse --short HEAD) tag_name=$(Build.BuildId)
+VERSION_TO_DOWNLOAD=$(curl -s
+"https://api.github.com/repos/CatalystCode/spk/releases/latest" | grep
+"tag_name" | sed -E 's/.*"([^"]+)".*/\1/') echo "Downloading SPK version
+$VERSION_TO_DOWNLOAD" && wget
+"https://github.com/CatalystCode/spk/releases/download/$VERSION_TO_DOWNLOAD/spk-linux"
+chmod +x ./spk-linux ./spk-linux deployment create  -n $(ACCOUNT_NAME) -k
+$(ACCOUNT_KEY) -t $(TABLE_NAME) -p $(PARTITION_KEY)  --p2 $(Build.BuildId)
+--hld-commit-id $latest_commit --env $(Build.SourceBranchName) --image-tag
+$tag_name
```
Comment (samiyaakhtar) [inline docs/service-introspection-onboarding.md]: Same here, new lines are going to cause this bash task to fail if the user just copy pastes it

**Review by samiyaakhtar — CHANGES_REQUESTED** [review]

**Comment by samiyaakhtar** [comment]
@edaena I think we already seem to have documentation in place for multi-stage pipelines, right?

File: docs/service-introspection-onboarding.md [hunk 3]
```diff
@@ -149,32 +149,59 @@ by service introspection.
...
+Paste the following task towards the end of your release step in the release
+pipeline in the Azure DevOps portal:
 
 ```yaml
-latest_commit=$(git rev-parse --short HEAD)
-VERSION_TO_DOWNLOAD=$(curl -s "https://api.github.com/repos/CatalystCode/spk/releases/latest" | grep "tag_name" | sed -E 's/.*"([^"]+)".*/\1/') 
-echo "Downloading SPK version $VERSION_TO_DOWNLOAD" && wget "https://github.com/CatalystCode/spk/releases/download/$VERSION_TO_DOWNLOAD/spk-linux"
-chmod +x ./spk-linux
-./spk-linux deployment create  -n $(ACCOUNT_NAME) -k $(ACCOUNT_KEY) -t $(TABLE_NAME) -p $(PARTITION_KEY)  --p2 $(Release.ReleaseId) --hld-commit-id $latest_commit --env $(Release.EnvironmentName) --image-tag $(Build.BuildId)
+latest_commit=$(git rev-parse --short HEAD) VERSION_TO_DOWNLOAD=$(curl -s
```
Comment (samiyaakhtar) [inline docs/service-introspection-onboarding.md]: New line between `VERSION_TO_DOWNLOAD` and `latest_commit=$(git rev-parse --short HEAD)`

File: docs/service-introspection-onboarding.md [hunk 4]
```diff
@@ -149,32 +149,59 @@ by service introspection.
...
+"https://api.github.com/repos/CatalystCode/spk/releases/latest" | grep
+"tag_name" | sed -E 's/.*"([^"]+)".*/\1/') echo "Downloading SPK version
```
Comment (samiyaakhtar) [inline docs/service-introspection-onboarding.md]: Newline before echo as well

File: docs/service-introspection-onboarding.md (hunk 1 above)
Comment (samiyaakhtar) [reply docs/service-introspection-onboarding.md]: @edaena I think this is still valid, may want to remove this diff altogether

**Review by samiyaakhtar — APPROVED** [review]

**Pull request merged**